src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -263,10 +263,12 @@ public CompiledScript compileInternal(Script script) {
             return compiled;
         }
 
+        String name = "";
```

Comment:
For this and the comment below about null -- this was two different lines of thinking, and I forgot to change the latter.  I was thinking for inline scripts I would leave the string empty.  Given Simon's suggestion of a toString method, I'm going to go back to null.